[error tracking] Add setup instructions for ET Backend standalone (#28374)
DataDog/documentation: content/en/error_tracking/backend/getting_started.md

  text: 'Error Tracking Issue States and Workflows'
- link: '/error_tracking/backend/exception_replay'
  tag: 'Documentation'
  text: 'Simplify production debugging with Datadog Exception Replay'
- link: "/error_tracking/explorer"
  tag: "Documentation"
  text: "Learn about the Error Tracking Explorer"
---

## Overview

[Error Tracking][1] processes errors collected by the Datadog Agent. Whenever an [error span][2] or span containing an error [span event][3] is collected, Error Tracking processes and groups it under an issue, or group of similar errors.

## Getting started

Follow the [in-app setup instructions][4] or choose an instrumentation approach to start collecting backend errors:

### Using Single Step Instrumentation

If you install or update a Datadog Agent with the **Enable APM Instrumentation** and **Error Tracking Standalone** options selected, the Agent is installed and configured to enable Backend Error Tracking. This allows you to automatically instrument your application, without any additional installation or configuration steps.

The following examples show how it works for each deployment type.

{{< tabs >}}
{{% tab "Linux host or VM" %}}

> For an Ubuntu host:

Reviewer: Should be the same across many, if not all, supported linux distros, not just ubuntu

1. Run the one-line installation command:

   ```shell
   DD_API_KEY=<YOUR_DD_API_KEY> DD_SITE="<YOUR_DD_SITE>" DD_APM_INSTRUMENTATION_ENABLED=host DD_APM_INSTRUMENTATION_LIBRARIES="java:1,python:2,js:5,dotnet:3,php:1" DD_APM_ERROR_TRACKING_STANDALONE=true DD_ENV=<AGENT_ENV> bash -c "$(curl -L https://install.datadoghq.com/scripts/install_script_agent7.sh)"
   ```

   Replace `<YOUR_DD_API_KEY>` with your [Datadog API key][5], `<YOUR_DD_SITE>` with your [Datadog site][6], and `<AGENT_ENV>` with the environment your Agent is installed on (for example, `staging`).

> 2. Start a new shell session.

Reviewer: I don't think starting a new shell session is needed

If you're using the Docker containerized Agent, set the following environment variables:
- `DD_CORE_AGENT_ENABLED=false`
- `DD_APM_ERROR_TRACKING_STANDALONE_ENABLED=true`

Here's an example of how you can include these settings in your Docker run command:

```shell
docker run -d --name datadog-agent \
           --cgroupns host \
           --pid host \
           -e DD_API_KEY=<DATADOG_API_KEY> \
           -e DD_CORE_AGENT_ENABLED=false \
           -e DD_APM_ERROR_TRACKING_STANDALONE_ENABLED=true \
           -v /var/run/docker.sock:/var/run/docker.sock:ro \
           -v /proc/:/host/proc/:ro \
           -v /opt/datadog-agent/run:/opt/datadog-agent/run:rw \
           -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
           gcr.io/datadoghq/agent:latest
```

{{% /tab %}}
{{% tab "Kubernetes" %}}

If you're deploying the Agent in Kubernetes, make the following changes in your Helm chart in addition to your Agent configuration:

```yaml
agents:
  containers:
    agent:
      env:
        - name: DD_CORE_AGENT_ENABLED
          value: "false"
datadog:
[...]
  processAgent:
    enabled: false
    containerCollection: false
[...]
  apm:
    errorTrackingStandalone:

>       enabled: true

Reviewer: should we recommend SSI here as well?

Author: I don't think so as it is in the "non-SSI"section. But it is indeed confusing, I'm really thinking of splitting this page into two (one for SSI setup, one for non-SSI - just like APM does)

---
title: Backend Error Tracking
aliases:
- /error_tracking/standalone_backend/getting_started
further_reading:
- link: '/error_tracking/issue_states/'
  tag: 'Documentation'
  text: 'Error Tracking Issue States and Workflows'
- link: '/error_tracking/backend/exception_replay'
  tag: 'Documentation'
  text: 'Simplify production debugging with Datadog Exception Replay'
- link: "/error_tracking/explorer"
  tag: "Documentation"
  text: "Learn about the Error Tracking Explorer"
---

## Overview

> [Error Tracking][1] processes errors collected by the Datadog Agent. Whenever an [error span][2] or span containing an error [span event][3] is collected, Error Tracking processes and groups it under an issue, or group of similar errors.

Reviewer 2: ```suggestion
[Error Tracking][1] processes errors collected by the Datadog Tracing Libraries. Whenever an [error] is collected, Error Tracking processes and groups it under an issue, or group of similar errors.
```

Reviewer 2: for standalone backend we probably shouldn't talk about spans

## Getting started

Follow the [in-app setup instructions][4] or choose an instrumentation approach to start collecting backend errors:

> 

Reviewer 2: We should probably list out and link to the two approaches

- Single Step Instrumentation
- Manual Instrumentation

### Using Single Step Instrumentation

> If you install or update a Datadog Agent with the **Enable APM Instrumentation** and **Error Tracking Standalone** options selected, the Agent is installed and configured to enable Backend Error Tracking. This allows you to automatically instrument your application, without any additional installation or configuration steps.

Reviewer 2: ```suggestion
Install or update a Datadog Agent with the **Enable APM Instrumentation** and **Error Tracking Standalone** options, to enable standalone Backend Error Tracking. This allows you to automatically instrument your application, without any additional installation or configuration steps.
```

   datadog:
     processAgent:
       enabled: false
       containerCollection: false
     apiKeyExistingSecret: datadog-secret
     site: <DATADOG_SITE>
     tags:
       - env:<AGENT_ENV>
     apm:
       errorTrackingStandalone:
         enabled: true
       # Required to enable Single-Step Instrumentation
       instrumentation:
         enabled: true
         libVersions:
           java: "1"
           dotnet: "3"
           python: "2"
           js: "5"
           php: "1"
   ```
   Replace `<DATADOG_SITE>` with your [Datadog site][10] and `<AGENT_ENV>` with the environment your Agent is installed on (for example, `env:staging`).

4. Run the following command to deploy the agent:
   ```shell
   helm install datadog-agent -f datadog-values.yaml datadog/datadog
   ```
5. After waiting a few minutes for the Datadog Cluster Agent changes to apply, restart your applications.

{{< /collapse-content >}}

[7]: https://v3.helm.sh/docs/intro/install/
[8]: https://kubernetes.io/docs/tasks/tools/install-kubectl/
[9]: https://app.datadoghq.com/organization-settings/api-keys
[10]: /getting_started/site/

{{% /tab %}}
{{< /tabs >}}

### Using Datadog tracing libraries

> To automatically instrument your application with Datadog libraries:

Reviewer 2: ```suggestion
To instrument your application with Datadog libraries:
```

```shell
docker run -d --name datadog-agent \
           --cgroupns host \
           --pid host \
           -e DD_API_KEY=<DATADOG_API_KEY> \
           -e DD_CORE_AGENT_ENABLED=false \
           -e DD_APM_ERROR_TRACKING_STANDALONE_ENABLED=true \
           -v /var/run/docker.sock:/var/run/docker.sock:ro \
           -v /proc/:/host/proc/:ro \
           -v /opt/datadog-agent/run:/opt/datadog-agent/run:rw \
           -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
           gcr.io/datadoghq/agent:latest
```

{{% /tab %}}
{{% tab "Kubernetes" %}}

If you're deploying the Agent in Kubernetes, make the following changes in your Helm chart in addition to your Agent configuration:

```yaml
agents:
  containers:
    agent:
      env:
        - name: DD_CORE_AGENT_ENABLED
          value: "false"
datadog:
[...]
  processAgent:
    enabled: false
    containerCollection: false
[...]
  apm:
    errorTrackingStandalone:
      enabled: true
```

{{% /tab %}}
{{< /tabs >}}

#### Instrument your application

Follow the relevant [documentation][14] to set up your application to send traces using one of the official Datadog tracing libraries.

> 

Reviewer 2: We probably want to add a doc page / link to a page about manually sending errors (through span events)

### Advanced options

> #### Enable Infrastructure Monitoring

Reviewer 2: I actually might add these as guides instead of the getting started for BE standalone
